nuxeo-core/nuxeo-core-io/src/main/java/org/nuxeo/ecm/core/io/marshallers/json/enrichers/AbstractJsonEnricher.java

```diff
@@ -56,15 +64,30 @@ public final boolean accept(Class<?> clazz, Type genericType, MediaType mediatyp
 
     @Override
     public void write(Enriched<EntityType> enrichable, JsonGenerator jg) {
-        try {
-            write(jg, enrichable.getEntity());
+        try (TokenBuffer tb = new TokenBuffer(MAPPER, false)) {
+            // Write to a temporary output in case of exception during write()
+            tb.writeStartObject();
+            write(tb, enrichable.getEntity());
+            tb.writeEndObject();
+            tb.flush();
+            // Add the complete, well-formed content to the real output
+            try (JsonParser parser = tb.asParser()) {
+                parser.nextToken(); // ignoring START_OBJECT
+                while (parser.nextToken() == FIELD_NAME) {
+                    jg.copyCurrentStructure(parser);
+                }
+                if (parser.currentToken() != END_OBJECT) {
+                    log.error("Enricher: {} failed on current token: {}, output to write: {}", name,
+                            parser.currentToken(), MAPPER.readTree(tb.asParser()));
```

Comment:
Can you try/catch readTree call?

Reply:
added it like so: https://github.com/nuxeo/nuxeo/pull/3799#discussion_r387688501